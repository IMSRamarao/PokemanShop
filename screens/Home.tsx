import {View, Text, ScrollView, StyleSheet} from 'react-native';
import React, {useEffect, useState} from 'react';
import Header from '../src/components/Header';
import Products from '../src/components/Products';

interface Products {
  id: number;
  title: string;
  price: number;
  description: string;
  category: string;
  image: string;
  rating: number;
}
[];

const Home = () => {
  const [product, setProduct] = useState<Products[]>([]);
  const [loading, setLoading] = useState(true);
  const fetchData = async () => {
    const resp = await fetch(
      'https://fakestoreapiserver.reactbd.com/amazonproducts',
    );
    const data = await resp.json();
    setProduct(data);
    setLoading(false);
  };
  useEffect(() => {
    fetchData();
  }, []);
  return (
    <View >
      {loading && (
        <View style={styles.loader}>
          <Text style={styles.loadingData}>Data is loading...</Text>
        </View>
      )}

      <ScrollView>
        <View style={styles.titleContainer}>
          <Text style={styles.title}>Pokeman Store</Text>
          <Text style={styles.app_Des}>
            This is Pokeman app to to get the data of pokeman
          </Text>
        </View>
        <View style={styles.list}>
          {product.map(item => (
            <Products key={item.id} item={item} />
          ))}
        </View>
      </ScrollView>
    </View>
  );
};

const styles = StyleSheet.create({
  loader: {
    height: '70%',
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center',
  },
  titleContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    textAlign: 'center',
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    padding: 10,
  },
  app_Des: {
    fontSize: 16,
    fontWeight: '400',
    padding: 10,
  },
  list: {
    width: '100%',
  },
  loadingData: {
    fontSize: 25,
    fontWeight: 'bold',
  },
});

export default Home;
