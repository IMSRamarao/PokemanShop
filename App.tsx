import React from 'react';
import Home from './screens/Home';
import {NavigationContainer} from '@react-navigation/native';
import {createNativeStackNavigator} from '@react-navigation/native-stack';
import SingleProducts from './screens/SingleProducts';
import Cart from './screens/Cart';
import {store} from './redux/store';
import {Provider} from 'react-redux';
import Payment from './screens/Payment';
import ShoppingCart from './src/components/ShoppingCart';
import TabNavigator from './navigation/TabNavigator';

const Stack = createNativeStackNavigator();

const App = () => {
  return (
    <Provider store={store}>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen
            name="Home"
            component={TabNavigator}
            options={({navigation}) => ({
              headerRight: () => <ShoppingCart navigation={navigation} />,
            })}
          />
          <Stack.Screen
            name="SingleProducts"
            component={SingleProducts}
            options={({navigation}) => ({
              headerRight: () => <ShoppingCart navigation={navigation} />,
            })}
          />
          <Stack.Screen
            name="Cart"
            component={Cart}
            options={({navigation}) => ({
              headerRight: () => <ShoppingCart navigation={navigation} />,
            })}
          />
          <Stack.Screen
            name="Payment"
            component={Payment}
            options={({navigation}) => ({
              headerRight: () => <ShoppingCart navigation={navigation} />,
            })}
          />
        </Stack.Navigator>
      </NavigationContainer>
    </Provider>
  );
};

export default App;
