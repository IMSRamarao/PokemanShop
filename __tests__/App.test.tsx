/**
 * @format
 */

import 'react-native';
import React from 'react';
import App from '../App';
import Home from '../screens/Home';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.useFakeTimers();
test('App snapshots', () => {
  const snap = renderer.create(<App />).toJSON();
  expect(snap).toMatchSnapshot();
});

test('Home snapshots', () => {
  const snap = renderer.create(<Home />).toJSON();
  expect(snap).toMatchSnapshot();
});
