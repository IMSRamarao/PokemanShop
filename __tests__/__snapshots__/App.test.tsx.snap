// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshots 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                undefined,
              ]
            }
          >
            <RNSScreenNavigationContainer
              style={
                {
                  "flex": 1,
                  "overflow": "hidden",
                }
              }
            >
              <RNSModalScreen
                activityState={2}
                collapsable={false}
                forwardedRef={[Function]}
                gestureResponseDistance={
                  {
                    "bottom": -1,
                    "end": -1,
                    "start": -1,
                    "top": -1,
                  }
                }
                onGestureCancel={[Function]}
                sheetAllowedDetents="large"
                sheetCornerRadius={-1}
                sheetExpandsWhenScrolledToEdge={true}
                sheetGrabberVisible={false}
                sheetLargestUndimmedDetent="all"
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "zIndex": 0,
                  }
                }
              >
                <View
                  accessibilityElementsHidden={false}
                  importantForAccessibility="auto"
                  style={
                    [
                      {
                        "backgroundColor": "rgb(242, 242, 242)",
                        "flex": 1,
                      },
                      [
                        {
                          "flex": 1,
                          "flexDirection": "column-reverse",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      {
                        "flex": 1,
                      }
                    }
                  >
                    <View>
                      <View
                        style={
                          {
                            "alignItems": "center",
                            "height": "70%",
                            "justifyContent": "center",
                            "textAlign": "center",
                          }
                        }
                      >
                        <Text
                          style={
                            {
                              "fontSize": 25,
                              "fontWeight": "bold",
                            }
                          }
                        >
                          Data is loading...
                        </Text>
                      </View>
                      <RCTScrollView>
                        <View>
                          <View
                            style={
                              {
                                "alignItems": "center",
                                "flex": 1,
                                "justifyContent": "center",
                                "textAlign": "center",
                              }
                            }
                          >
                            <Text
                              style={
                                {
                                  "fontSize": 20,
                                  "fontWeight": "bold",
                                  "padding": 10,
                                }
                              }
                            >
                              Pokeman Store
                            </Text>
                            <Text
                              style={
                                {
                                  "fontSize": 16,
                                  "fontWeight": "400",
                                  "padding": 10,
                                }
                              }
                            >
                              This is Pokeman app to to get the data of pokeman
                            </Text>
                          </View>
                          <View
                            style={
                              {
                                "width": "100%",
                              }
                            }
                          />
                        </View>
                      </RCTScrollView>
                    </View>
                  </View>
                </View>
              </RNSModalScreen>
            </RNSScreenNavigationContainer>
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": true,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignContent": "center",
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    {
                      "backgroundColor": "white",
                      "color": "#673ab7",
                      "padding": 20,
                      "textAlign": "center",
                    }
                  }
                >
                  Amazon
                </Text>
              </View>
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": undefined,
                    "expanded": undefined,
                    "selected": undefined,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "alignContent": "center",
                    "flex": 1,
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    {
                      "backgroundColor": "white",
                      "color": "#222",
                      "padding": 20,
                      "textAlign": "center",
                    }
                  }
                >
                  PokeMan
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Home"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      >
        <RNSScreenStackHeaderSubview
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          type="right"
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  {
                    "alignItems": "center",
                    "backgroundColor": "black",
                    "borderRadius": 100,
                    "height": 20,
                    "justifyContent": "center",
                    "left": 22,
                    "position": "absolute",
                    "right": 0,
                    "top": 10,
                    "width": 20,
                    "zIndex": 2000,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "white",
                      "fontSize": 16,
                      "fontWeight": "bold",
                    }
                  }
                >
                  0
                </Text>
              </View>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="25"
                bbWidth="25"
                color="#131921"
                data-slot="icon"
                fill="currentColor"
                focusable={false}
                height={25}
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    {
                      "marginRight": 10,
                    },
                    {
                      "flex": 0,
                      "height": 25,
                      "width": 25,
                    },
                  ]
                }
                tintColor="#131921"
                vbHeight={24}
                vbWidth={24}
                width={25}
              >
                <RNSVGGroup
                  fill={
                    {
                      "type": 2,
                    }
                  }
                  propList={
                    [
                      "fill",
                    ]
                  }
                >
                  <RNSVGPath
                    d="M2.25 2.25a.75.75 0 0 0 0 1.5h1.386c.17 0 .318.114.362.278l2.558 9.592a3.752 3.752 0 0 0-2.806 3.63c0 .414.336.75.75.75h15.75a.75.75 0 0 0 0-1.5H5.378A2.25 2.25 0 0 1 7.5 15h11.218a.75.75 0 0 0 .674-.421 60.358 60.358 0 0 0 2.96-7.228.75.75 0 0 0-.525-.965A60.864 60.864 0 0 0 5.68 4.509l-.232-.867A1.875 1.875 0 0 0 3.636 2.25H2.25ZM3.75 20.25a1.5 1.5 0 1 1 3 0 1.5 1.5 0 0 1-3 0ZM16.5 20.25a1.5 1.5 0 1 1 3 0 1.5 1.5 0 0 1-3 0Z"
                    fill={
                      {
                        "payload": 4278190080,
                        "type": 0,
                      }
                    }
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </View>
        </RNSScreenStackHeaderSubview>
      </RNSScreenStackHeaderConfig>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;

exports[`Home snapshots 1`] = `
<View>
  <View
    style={
      {
        "alignItems": "center",
        "height": "70%",
        "justifyContent": "center",
        "textAlign": "center",
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 25,
          "fontWeight": "bold",
        }
      }
    >
      Data is loading...
    </Text>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
            "textAlign": "center",
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 20,
              "fontWeight": "bold",
              "padding": 10,
            }
          }
        >
          Pokeman Store
        </Text>
        <Text
          style={
            {
              "fontSize": 16,
              "fontWeight": "400",
              "padding": 10,
            }
          }
        >
          This is Pokeman app to to get the data of pokeman
        </Text>
      </View>
      <View
        style={
          {
            "width": "100%",
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
